engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -460,7 +460,26 @@ protected void restoreOutput(PrintStream originalStream)
     }
   }
 
-  AppPackage newAppPackageInstance(File f) throws IOException, ZipException
+  AppPackage newAppPackageInstance(URI uri, boolean suppressOutput) throws IOException
+  {
+    PrintStream outputStream = suppressOutput ? suppressOutput() : null;
+    try {
+      final String scheme = uri.getScheme();
+      if (scheme == null || scheme.equals("file")) {
+        return new AppPackage(new FileInputStream(new File(expandFileName(uri.getPath(), true))), true);
+      } else {
```

Comment:
Why not do special handling for hdfs only when scheme is "dfs type" such as hdfs or s3 types. In other cases go with default url handlers using uri.toURL().openStream()? This will be able to handle other sources such as http and ftp.